README.md

```diff
@@ -1,10 +1,384 @@
+
+{{#hw.device-type.storage.internal}}
+4. The device should appear on you dashboard in a configuring state. {{import partial="description-of-internal-process" combination="sw.image-writer"}}. {{import partial="visual-appearance-when-off" combination="hw.device-type"}}
+
+5. {{import partial="remove-install-media" combination="hw.device-type"}}. {{import partial="boot-internal" combination="hw.device-type"}}
+{{/hw.device-type.storage.internal}}
+
+6. Your device should appear here in the IDLE state in 30 seconds or so. Have fun!
+
+**Troubleshooting:** If, upon boot, the device LED is blinking in groups of four, it is an indication that the device cannot connect to the internet. Please ensure the network adapter is functional.
+
+{{#compare hw.device-type.media.installation "!==" "dfu"}}
+**Pro tip:** You can repeat the initialisation steps for any amount of {{hw.device-type.name}}'s you have available, using the same {{hw.device-type.media.installation}}.
+{{/compare}}
+```
+
+Contracts can be rendered with contrato using the [buildTemplate](https://balena-io.github.io/contrato/modules/contrato.html#buildtemplate) function
+
+```ts
+import { buildTemplate } from 'contrato';
+import { promises as fs } from 'fs';
+
+// Reproduce the blueprint over the universe
+const contexts = blueprint.reproduce(universe);
+const template = await fs.readFile('instructions.tpl', 'utf8');
+
+for (const context of contexts) {
+	console.log(await buildTemplate(template, context));
+}
+```
+
+### Partials
+
+Each contract or combination of contracts may have partials associated with it that may be assembled into text artifacts with the template system.
+
+The convention is to have a directory of contracts with the following structure:
+
+```
+<type combinations>/<slug combinations>/contract.json
+<type combinations>/<slug combinations>/partial1.tpl
+<type combinations>/<slug combinations>/partial2.tpl
+<type combinations>/<slug combinations>/partialN.tpl
+```
+
+The types and slugs fragments of the path might be one entity, or a set of entities separated by a + sign. For example:
+
+```
+sw.os+arch.sw/debian+amd64/installation.tpl
+hw.device-type/ts4900/remove-install-media.tpl
+```
+
+The combination section defines the types of contracts that come into play for a particular partials subtree, separated by a `+` symbol. If the combination is sw.os+arch.sw, then it means that the subtree will take into account the combination of operating systems and architectures. Note that there can be combinations of a single type.
+
+The combination instantiation section defines a subtree for a specific set of contracts that match the combination type. If the combination is `sw.os+arch.sw`, a valid combination instantiation can be `debian+amd64`, which is the subtree that will be selected when matching the Debian GNU/Linux contract with the amd64 architecture contract.
```

Comment:
I'm confused by the difference between this sentence and the one above?

> The combination section defines the types of contracts that come into play for a particular partials subtree, separated by a `+` symbol. If the combination is sw.os+arch.sw, then it means that the subtree will take into account the combination of operating systems and architectures. Note that there can be combinations of a single type.

> The combination instantiation section defines a subtree for a specific set of contracts that match the combination type. If the combination is `sw.os+arch.sw`, a valid combination instantiation can be `debian+amd64`, which is the subtree that will be selected when matching the Debian GNU/Linux contract with the amd64 architecture contract.

Reply:
Let me re-read this, I pretty much copy pasted this from the original spec

Reply:
So, what it says its that the combination of contracts go in a folder that declares what types you are combining and then the instances of the types. So if you want to store a partial for `debian+amd64` (the instances), the partials need to go into a folder `sw.os+arch.sw/debian+amd64/`

Reply:
It is a pretty verbose way to explain it which probably adds to the confusion, I'll try to re-write it

Reply:
I made some slight tweaks, but I think the descriptions in each paragraph are accurate.